modules/basegl/src/processors/tuberendering.cpp

```diff
@@ -144,6 +144,8 @@ void TubeRendering::process() {
         return false;
     };
 
+    utilgl::CullFaceState cullstate(GL_FRONT);
```

Comment:
I guess it doesn't make a difference since the shader doesn't care, but this will remove the front faces, not the back faces. It itches my eyes a bit, I would have used `GL_BACK` instead. :)

Reply:
Why is this needed? will this cull the output from the geometry shader? why would that be needed? should have a comment... why do we generate triangels in the geometry shader that we don't want. would it not be better not to create those to begin with...

Reply:
In the geometry shader, we generate a bounding box with all 6 sides. The frag shader does not consider if the current fragment is a front- or backface. The ray-cylinder intersection test will give the same result for both, hence the z-fighting in #897 . 

I guess we could try to figure out which sides are visible given a view direction and only emit those, but I doubt it will be faster (nor slower) than the face culling.

Reply:
But yes, A comment why this is needed might be good.

Reply:
go with the comment, that is not obvious...

Reply:
The alternative would be to use `if (!gl_FrontFacing) discard;` in the fragment shader. See [khronos gl_FrontFacing](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/gl_FrontFacing.xhtml).

Reply:
This is more expensive though
"Face culling allows non-visible triangles of closed surfaces to be removed before expensive Rasterization and Fragment Shader operations."

Reply:
would be even better not to create them to begin with...